cupy_backends/cuda/libs/cusparse.pyx

```diff
@@ -1384,6 +1385,22 @@ cdef dict STATUS = {
 }
 
 
+cdef dict HIP_STATUS = {
```

Comment:
having hip related stuff in `cupy_backends/cuda` scares me a bit, but I can't think of any other place to hold this ...

Reply:
Yeah, unfortunately they need to stay here. Perhaps `cupy_backends/cuda` should be renamed to `cupy_backends/impl` 😅 We have done this for hipBLAS/rocSOLVER/etc anyway.

Reply:
yeah, I can only think of creating .pyx files in hip with all these redeclarations and change the module pointers for this functions python-side, but I am not sure if that would even work.

Reply:
Sounds like we need the PRs https://github.com/cupy/cupy/pull/4395 etc would help...